src/types.js

```diff
@@ -102,13 +102,15 @@ export type Account = {|
  */
 export type Identity = $Diff<Auth, { apiKey: string }>;
 
+export type EmojiType = 'unicode' | 'image';
+
 /** An aggregate of all the reactions with one emoji to one message. */
 export type AggregatedReaction = {|
   code: string,
   count: number,
   name: string,
   selfReacted: boolean,
-  type: ReactionType,
+  type: EmojiType,
```

Comment:
Ya, more code needs to be changed. We send back this to server when reaction is clicked (to add/remove). 

But wondering why flow doesn't complained 🤔 

```
reactionType: requireAttribute(target, 'data-type'),
```

^ this is resolved to `string`

```
type MessageListEventReaction = {|
  type: 'reaction',
  messageId: number,
  name: string,
  code: string,
  reactionType: string,
  voted: boolean,
|};
```

And at the time of calling api, we consider it as `string`
```
export default (
  auth: Auth,
  messageId: number,
  reactionType: string,
  emojiCode: string,
  emojiName: string,
): Promise<ApiResponse> =>
  apiDelete(auth, `messages/${messageId}/reactions`, {
    reaction_type: reactionType,
    emoji_code: emojiCode,
    emoji_name: emojiName,
  });
```

So flow doesn't complains.

I thinks it will be hard to apply flow here, as data is coming from attribute, which is string.

Should we add new attribute at the time of aggregating reactions?

Reply:
Replied in the [chat thread](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M3847.20emoji.3A.20Use.20parseUnicodeEmojiCode.20to.20render.20unicode.20em/near/819793) you started (good idea!)